### PR TITLE
cf 245h yo 🤡 

### PR DESCRIPTION
**Pull request by envyaims**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/cf-245H.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+id: cf-245H
+source: CF
+title: Queries for Number Of Palindromes
+author: Chongtian Ma
+---
+## Implementation
+
+Note this is a bit different from the implentation outlined in the official
+editorial. This loops over the start and end of the substrings rather than the
```
Comment (SansPapyrus683) [inline solutions/cf-245H.mdx:10]: link official

File: solutions/cf-245H.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+int main() {
+	cin.tie(0) -> sync_with_stdio(false);
+	string s;
+	cin >> s;
+	int n = s.length();
+	vector<vector<int>> dp(n, vector<int>(n));
+	vector<vector<bool>> pal(n, vector<bool>(n));
+	// dp[i][j] = number of palidrome substrings in [i,j]
+	// pal[i][j] = true if [i,j] is a palindrome
```
Comment (SansPapyrus683) [inline solutions/cf-245H.mdx:35]: put above the relevant variables

File: solutions/cf-245H.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	for (int i = n - 1; i >= 0; i--) {
+		// base case: strings of length 1 are always a palindrome
+		dp[i][i] = pal[i][i] = 1;
+		for (int j = i + 1; j < n; j++) {
+			/*
+			 We add a character before start and after the end.
+			 If previously was a palindrome and the new characters
+			 are the same, then it is still a palindrome.
+			*/
```
Comment (SansPapyrus683) [inline solutions/cf-245H.mdx:45]:
```suggestion
			/*
			 * We add a character before start and after the end.
			 * If previously was a palindrome and the new characters
			 * are the same, then it is still a palindrome.
			*/

File: solutions/cf-245H.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+						&& (s[i] == s[j]);
+
+			// number of palindromes using inclusion - exclusion
+			dp[i][j] = dp[i + 1][j] + dp[i][j - 1]
+					    - dp[i + 1][j - 1] + pal[i][j];
+		}
+	}
+
+	int q; cin >> q;
+	while (q--) {
```
Comment (SansPapyrus683) [inline solutions/cf-245H.mdx:56]:
```suggestion
	int q;
	cin >> q;
	for (int i = 0; i < q; i++) {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/cf-245H.mdx (hunk 1 above)
Comment (envyaims) [reply solutions/cf-245H.mdx:10]: i forgor 🤡

**Review by SansPapyrus683 — APPROVED** [review]
sure

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**